libpromises/evalfunction.c

```diff
@@ -923,6 +923,148 @@ static FnCallResult FnCallGetUsers(ARG_UNUSED EvalContext *ctx, ARG_UNUSED const
 
 /*********************************************************************/
 
+
+#if defined(HAVE_GETPWENT) && !defined(__ANDROID__)
+
+static FnCallResult FnCallFindLocalUsers(EvalContext *ctx, ARG_UNUSED const Policy *policy, const FnCall *fp, const Rlist *finalargs)
+{
+    bool allocated = false;
+    JsonElement *json = VarNameOrInlineToJson(ctx, fp, finalargs, false, &allocated);
+    
+    // we failed to produce a valid JsonElement, so give up
+    if (json == NULL)
+    {
+        Log(LOG_LEVEL_VERBOSE,
+            "Cannot produce valid JSON from the argument '%s' of the function '%s'",
+            fp->name, RlistScalarValueSafe(finalargs));
+        return FnFailure();
+    }
+    else if (JsonGetElementType(json) != JSON_ELEMENT_TYPE_CONTAINER)
+    {
+        Log(LOG_LEVEL_VERBOSE, "Function '%s', argument '%s' was not a data container or list",
+            fp->name, RlistScalarValueSafe(finalargs));
+            JsonDestroyMaybe(json, allocated);
+        return FnFailure();
+    }
+   
+    JsonElement *parent = JsonObjectCreate(10);
+    setpwent();
+    struct passwd *pw;
+    while ((pw = getpwent()))
+    {
+        JsonIterator iter = JsonIteratorInit(json);
+        bool canAddToJSON = true;
+        JsonElement *element = JsonIteratorNextValue(&iter);
+        while (element != NULL)
+        {
+            const char *field = JsonPrimitiveGetAsString(element);
+            const Rlist *tuple = RlistFromSplitString(field, '=');
+            const char *attribute = RlistScalarValue(tuple);
+
+            if (tuple->next == NULL)
+            {
+                Log(LOG_LEVEL_ERR, "Function %s couldn't parse %s", fp->name, field);
+                return FnFailure();
+            }
+            const char *value = RlistScalarValue(tuple->next);
+
+            if (StringEqual(attribute, "name"))
+            {
+                if (!StringMatchFull(value, pw->pw_name))
+                {
+                    canAddToJSON = false;
+                }
+            }
+            else if (StringEqual(attribute, "uid"))
+            {
+                int string_size = PRINTSIZE(pw->pw_uid);
+                char *uid_string = xmalloc(sizeof(char)*(string_size+1));
+
+                int val = sprintf(uid_string, "%u", pw->pw_uid);
```

Comment:
This variable is never used unless you compile in debug mode. Remember that asserts are stripped from the program unless debug mode is activated. You can silence the warning by adding the following macro in front:

```suggestion
                NDEBUG_UNUSED int val = sprintf(uid_string, "%u", pw->pw_uid);
```